crypto/pkcs11key/key_test.go

```diff
@@ -30,14 +41,39 @@ func (c mockCtx) FindObjectsFinal(sh pkcs11.SessionHandle) error {
 }
 
 func (c mockCtx) FindObjectsInit(sh pkcs11.SessionHandle, temp []*pkcs11.Attribute) error {
+	currentSearch = temp
 	return nil
 }
 
 func (c mockCtx) FindObjects(sh pkcs11.SessionHandle, max int) ([]pkcs11.ObjectHandle, bool, error) {
-	return []pkcs11.ObjectHandle{privateKeyHandle}, true, nil
+	for _, a := range currentSearch {
+		// We search private keys using CKA_LABEL
+		if a.Type == pkcs11.CKA_LABEL {
+			if a.Value[0] == 0x72 { // Label start with 'r'
```

Comment:
This will be in the next push. I also changed other sequences of if/else if for switch/case. With one small utility function I reduced the several functions shorter. I removed several magic values, replacing them with more explicit constants.